### PR TITLE
remove fluentd-gcp addon

### PR DESCRIPTION
**Pull request by coffeepac**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

removes fluentd-gcp from cluster/addons

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
fluentd-gcp addon has moved from kubernetes/kubernetes to kubernetes-sigs/instrumentation-addons
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
small part of https://github.com/kubernetes/kubernetes/issues/78995
```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -2734,7 +2748,7 @@ EOF
   fi
   if [[ "${ENABLE_NODE_LOGGING:-}" == "true" ]] && \
      [[ "${LOGGING_DESTINATION:-}" == "gcp" ]]; then
-    setup-addon-manifests "addons" "fluentd-gcp"
+    setup-addon-manifests-remote "addons" "fluentd-gcp" "https://github.com/coffeepac/instrumentation-addons/tree/pc-move-fluent-gcp/fluentd-gcp"
```
Comment (monotek) [inline cluster/gce/gci/configure-helper.sh:2751]: contains coffeepac in the url

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -2282,6 +2282,20 @@ function setup-addon-manifests {
...
+# $3: remote manifest repository
+function setup-addon-manifests-remote {
+  local -r dst_dir="/etc/kubernetes/$1/$2"
+  local -r local_addon_dir="/tmp/$2"
+
+  git clone $3 ${local_addon_dir}
+
+  copy-manifests "${local_addon_dir}/$2" "${dst_dir}"
+}
+
```
Comment (monotek) [inline cluster/gce/gci/configure-helper.sh:2298]: nice. i'll borrow that for the fluentd-elasticsaerch addon.

**Review by monotek — CHANGES_REQUESTED** [review]

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (coffeepac) [reply cluster/gce/gci/configure-helper.sh:2751]: yep!  the PR to create this hasn't been merged yet.  I'm trying to make sure the `setup-addon-manifests-remote` function works in CI.  I'll swap it over to the correct fork when the other thing is mered.

**Comment by coffeepac** [comment]
/hold

we shouldn't merge this until https://github.com/kubernetes-sigs/instrumentation-addons/pull/2 is merged so I can use the instrumentation-addons repo instead of my fork of it

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (ehashman) [reply cluster/gce/gci/configure-helper.sh:2745]: Please update.

Comment (ehashman) [reply cluster/gce/gci/configure-helper.sh:2745]: This still needs to be updated to the new k8s repo @coffeepac

File: cluster/gce/gci/configure-helper.sh [hunk 3]
```diff
@@ -2728,7 +2742,7 @@ EOF
   fi
   if [[ "${ENABLE_NODE_LOGGING:-}" == "true" ]] && \
      [[ "${LOGGING_DESTINATION:-}" == "gcp" ]]; then
-    setup-addon-manifests "addons" "fluentd-gcp"
+    setup-addon-manifests-remote "addons" "fluentd-gcp" "https://github.com/coffeepac/instrumentation-addons/tree/pc-move-fluent-gcp/fluentd-gcp"
```
Comment (liggitt) [inline cluster/gce/gci/configure-helper.sh:2745]: I'm a pretty strong -1 on adding remote addon installation to this script. See discussion in https://github.com/kubernetes/kubernetes/pull/101039#pullrequestreview-636921064

**Comment by dims** [comment]
This PR is older than 4 weeks and needs as rebase.

Please rebase the PR against latest master and reopen if still needed.

/close

**Pull request closed without merging**